scripts/react-compiler-compliance-check.ts

```diff
@@ -94,36 +131,25 @@ function runCompilerHealthcheck(src?: string): CompilerResults {
         cwd: process.cwd(),
     });
 
-    // Parse and then normalize via Set/Map to ensure true uniqueness
-    const parsed = parseCombinedOutput(output);
+    const results = parseHealthcheckOutput(output);
+    return results;
+}
 
-    // Use Set to deduplicate success entries
-    const successSet = new Set(parsed.success);
+function parseHealthcheckOutput(output: string): CompilerResults {
+    const lines = output.split('\n');
 
-    // Use Map keyed by unique file key to deduplicate failures
-    const failureMap = new Map<string, CompilerFailure>();
-    parsed.failures.forEach((failure) => {
-        const key = getUniqueFileKey(failure);
-        // Prefer the first occurrence that has a reason
-        const existing = failureMap.get(key);
-        if (!existing) {
-            failureMap.set(key, failure);
-            return;
-        }
-        if (!existing.reason && failure.reason) {
-            failureMap.set(key, failure);
-        }
-    });
+    const initialResults: CompilerResults = {
+        success: new Set<string>(),
+        failures: new Map<string, CompilerFailure>(),
+    };
 
-    return {success: successSet, failures: failureMap};
-}
+    const verboseResults = parseVerboseOutput(lines, initialResults);
```

Comment:
Having this intermediate state from `rawOutput -> verboseOutput -> finalOutput` feels kind of overcomplicated. Can we just do `rawOutput -> finalOutput`